### DISCUSSION TITLE
issue: watch API does't update values properly in combination with reset API

### DISCUSSION THREAD

**Discussion by d-soldatov** [react-hook-form/react-hook-form#8799] [General] [answered]
### Version Number

7.34.0

### Codesandbox/Expo snack

https://codesandbox.io/s/react-hook-form-reset-v7-forked-1nb7jo?file=/src/index.js

### Steps to reproduce

1. Go to form
2. Wait until the promise resolves (HTTP request simulation) and values are set
3. See that values were changed, but `watch` API returns only partial fields.
4. Click the `Re-render` button and see that the watch API returns correct values on the next render

NOTE: Uncomment `Devtools` and it works just fine with them enabled

### Expected behaviour

watch API updates values correctly

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
could you remove all other deps, please? so the issue is isolated, hookform Devtool use `useFormContext` inside which subscribe all formStates.

**Reply by d-soldatov** [reply]
Check it out: https://codesandbox.io/s/react-hook-form-reset-v7-forked-5sefug?file=/src/index.js

**Reply by bluebill1049** [reply]
This is due to `shouldUnregister: true,`

```
{allFields?.type === "tags" && (
  <>
    <input type="text" readOnly {...register("selectTags")} />

    <br />

    <pre style={{ color: "crimson" }}>
      {JSON.stringify(
        typeof allFields?.selectTags === "undefined"
          ? "Why selectTags is undefined in watch API?"
          : allFields?.selectTags,
        null,
        2
      )}
    </pre>
  </>
)}
```

the above condition will result in component get unmount and unregister, i would recommend to use `unregister` function manually for your use case.

**Comment by d-soldatov** [comment, marked as answer]
Thanks, @bluebill1049 you are right there. `shouldUnregister` is the key point. Just a weird thing that it works differently when Devtool is enabled/disabled

**Reply by bluebill1049** [reply]
Probably due to dev tool is flush down (more) re-renders
